File: C/Exercícios Udemy/diagonal_negativos.c
#include <stdio.h>
#define N 10

int main() {
    int n, i, j, m[N][N], v[N], negativos = 0;
    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &m[i][j]);
            if (i == j) v[j] = m[i][j];
            if (m[i][j] < 0) negativos++; 
        }
    }
    printf("DIAGONAL PRINCIPAL:\n");
    for (i = 0; i < n; i++) {
        printf("%d ", v[i]);
    }
    printf("\n");
    printf("QUANTIDADE DE NEGATIVOS = %d\n", negativos);
    return 0;
}

File: Java/Exercícios Udemy/diagonal_negativos.java
import java.util.Locale;
import java.util.Scanner;

public class diagonal_negativos {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        int[][] m;
        int[] v;
        int n, i, j, negativos = 0;
        System.out.println("Qual a ordem da matriz? ");
        n = sc.nextInt();
        m = new int[n][n];
        v = new int[n];
        for (i = 0; i < n; i++) {
            for (j = 0; j < n; j++) {
                System.out.println("Elemento [" + i + "," + j + "]: ");
                m[i][j] = sc.nextInt();
                if (i == j) v[j] = m[i][j];
                if (m[i][j] < 0) negativos++;
            }
        }
        System.out.println("DIAGONAL PRINCIPAL:");
        for (i = 0; i < n; i++) {
            System.out.println(v[i] + " ");
        }
        System.out.println("QUANTIDADE DE NEGATIVOS = " + negativos);
        sc.close();
    }
}
